File: 1666-make-the-string-great/make-the-string-great.java
class Solution {
    public String makeGood(String s) {

        Stack<Character> stack = new Stack<>();

        for(char curr : s.toCharArray()){

            if(!stack.isEmpty() && curr!=stack.peek() && Character.toUpperCase(curr)== Character.toUpperCase(stack.peek())) stack.pop();
            
            else stack.push(curr);
        }
        char[] res = new char[stack.size()];

        for(int i=stack.size()-1 ;i>=0;i--){

            res[i] = stack.pop();
        }
        return new String(res);
    }
}

File: 1666-make-the-string-great/make-the-string-great.js
/**
 * @param {string} s
 * @return {string}
 */
var makeGood = function(s) {

    let stack = [];
   

    for(let curr of s){
        let top = stack[stack.length-1];

        if (stack.length > 0 && curr!==top && 
        curr.toUpperCase() === top.toUpperCase()) stack.pop();

        else stack.push(curr);


    }
    return stack.join('');
    
};